yt_dlp/extractor/twitter.py

```diff
@@ -503,90 +607,114 @@ def add_thumbnail(name, size):
+                }
+            elif card_name == 'unified_card':
+                media_entities = self._parse_json(get_binding_value('unified_card'), twid)['media_entities']
+                media = traverse_obj(media_entities, ..., expected_type=dict, get_all=False)
+                return extract_from_video_info(media)
+            # amplify, promo_video_website, promo_video_convo, appplayer,
+            # video_direct_message, poll2choice_video, poll3choice_video,
+            # poll4choice_video, ...
+            else:
+                is_amplify = card_name == 'amplify'
+                vmap_url = get_binding_value('amplify_url_vmap') if is_amplify else get_binding_value('player_stream_url')
+                content_id = get_binding_value('%s_content_id' % (card_name if is_amplify else 'player'))
+                formats, subtitles = self._extract_formats_from_vmap_url(vmap_url, content_id or twid)
+                self._sort_formats(formats)
+
+                thumbnails = []
+                for suffix in ('_small', '', '_large', '_x_large', '_original'):
+                    image = get_binding_value('player_image' + suffix) or {}
+                    image_url = image.get('url')
+                    if not image_url or '/player-placeholder' in image_url:
+                        continue
+                    thumbnails.append({
+                        'id': suffix[1:] if suffix else 'medium',
+                        'url': image_url,
+                        'width': int_or_none(image.get('width')),
+                        'height': int_or_none(image.get('height')),
+                    })
+
+                return {
+                    'formats': formats,
+                    'subtitles': subtitles,
+                    'thumbnails': thumbnails,
+                    'duration': int_or_none(get_binding_value(
+                        'content_duration_seconds')),
+                }
+
+        entries = []
+        media_path = ((None, 'quoted_status'), 'extended_entities', 'media', ...)
+        medias = traverse_obj(status, media_path, expected_type=dict, default=[])
+        for media in medias:
+            if not media or media.get('type') == 'photo':
+                continue
+
+            data = extract_from_video_info(media)
+            if data:
+                entries.append({**info, **data, 'display_id': twid})
+            continue
+
+        card = status.get('card')
+        if card:
+            data = extract_from_card_info(card)
+            if data:
+                entries.append({**info, **data, 'display_id': twid})
+
+        if not entries:
+            expanded_url = traverse_obj(status, ('entities', 'urls', 0, 'expanded_url'))
+            if not expanded_url or expanded_url == url:
+                raise ExtractorError('No video could be found in this tweet')
+
+            transparent_result = self.url_result(expanded_url, transparent=True, display_id=twid, **info)
```

Comment:
are you sure this should be transparent? Does this generally get passed to generic? If it's being passed to another extractor, this probably shouldn't be transparent

Reply:
I have not thought about the implementation. This idea was taken from @dirkf's patch directly. I will reinvestigate.

Reply:
This was a mistake. The reason this worked is that the parameter for `InfoExtractor.url_result` was called `url_transparent` instead of `transparent`, so it kept the `_type` of `url`.

Reply:
hm.. Interesting, I thought the tests would complain if given a wrong field 🤔. I'll have to investigate (unrelated to this PR)

Reply:
This case got never hit in the tests since it was a fallback and we are now successfully extracting every tweets media. I don't know if this code makes sense at all, so I will have to take a look at the source response to make sense of it.

Reply:
For info, I had to set `url_transparent` in just these cases to pass the tests with fewest changes:
* `card_name == 'periscope_broadcast'`
* `card_name == 'broadcast'`